authserver/db/models/models.py

```diff
@@ -339,3 +304,96 @@ class AuthorizedClient(db.Model):
     user_id = db.Column(db.String, db.ForeignKey('users.id'), primary_key=True)
     client_id = db.Column(db.String, primary_key=True)
     authorized = db.Column(db.Boolean, nullable=False, default=False)
+
+
+class Scope(db.Model):
+    """OAuth2 scopes associated with users and clients.
+
+    This class provides an abstraction of an OAuth2 scope. The OAuth2Client entity has a scope
+    field provided by its OAuth2ClientMixin class; however this level of abstraction is
+    needed in order to expose this to clients in an easier way as well as to facilitate the linkage
+    of scopes between clients and users.
+
+    """
+
+    __tablename__ = 'oauth2_scopes'
+    __table_args__ = (db.UniqueConstraint('scope'), )
+
+    id = db.Column(db.String, primary_key=True)
+    scope = db.Column(db.String, unique=True, nullable=False)
+    description = db.Column(db.String)
+    date_created = db.Column(db.TIMESTAMP)
+    date_last_updated = db.Column(db.TIMESTAMP)
+
+    def __init__(self, scope, description):
+        self.id = str(uuid4()).replace('-', '')
+        self.scope = scope
+        self.description = description
+        self.date_created = datetime.utcnow()
+        self.date_last_updated = datetime.utcnow()
+
+    def __str__(self):
+        return self.id
+
+
+class ScopeSchema(ma.SQLAlchemySchema):
+    """Scope schema
+
+    A marshmallow schema for validating the Scope model.
+    """
+
+    class Meta:
+        ordered = True
+        model = Scope
+
+    id = ma.auto_field(dump_only=True)
+    scope = ma.auto_field(required=True)
+    description = ma.auto_field(required=True)
+    date_created = ma.auto_field(dump_only=True)
+    date_last_updated = ma.auto_field(dump_only=True)
+
+
+class AuthorizedScope(db.Model):
+    """Map scopes to roles.
```

Comment:
Admittedly, I am having some trouble with the name `AuthorizedScope`. 

The `AuthorizedScope` table strikes me as a beefed-up join table, and its name does not really communicate that. I think a conventional relational name would be `RoleScope` or `RoleScopes`. Is there a reason why you prefer `AuthorizedScope`?